libraries/core_libs/network/src/tarcap/packets_handlers/get_pbft_sync_packet_handler.cpp

```diff
@@ -39,6 +42,13 @@ void GetPbftSyncPacketHandler::process(const PacketData &packet_data,
     throw MaliciousPeerException(err_msg.str());
   }
 
+  if (is_light_node_ && height_to_sync + light_node_history_ <= my_chain_size) {
+    std::ostringstream err_msg;
+    err_msg << "Peer " << packet_data.from_node_id_ << " request syncing period start at " << height_to_sync
+            << ". Light node does not have the data " << my_chain_size;
+    throw MaliciousPeerException(err_msg.str());
```

Comment:
I think sender should not know the peer is a light node and don't have the block before sending request. Do we need to mark sender as malicous here?

Reply:
Sender should know this since it is being sent as part of status message